Sources/Target/Common/ProcessBase.cpp

```diff
@@ -288,6 +284,26 @@ ErrorCode ProcessBase::afterResume() {
   return kSuccess;
 }
 
+SoftwareBreakpointManager *ProcessBase::softwareBreakpointManager() const {
+#if defined(ARCH_ARM) || defined(ARCH_ARM64)
+  typedef Architecture::ARM::SoftwareBreakpointManager SWBPMType;
+#elif defined(ARCH_X86) || defined(ARCH_X86_64)
+  typedef Architecture::X86::SoftwareBreakpointManager SWBPMType;
+#endif
+  static SWBPMType softwareBreakpointManager(const_cast<ProcessBase *>(this));
```

Comment:
I don't like this. This implies I cannot debug multiple processes at once (with one ds2 instance). Why can't this be a member variable?